### DISCUSSION TITLE
Make CKAN more "flask friendly"

### DISCUSSION THREAD

**Discussion by pdelboca** [ckan/ckan#6776] [General]
Hi everyone!

While working with CKAN I notice that we have added a "CKAN layer" on top of Flask framework. Examples:
 - `ckan run` instead of `flask run`
 - `from ckan.common import session` instead of `from flask import session`
 - `CKANConfig` instead of `app.config`
 - CKANTestApp instead of Flask's test_client
 - etc

I understand that there are some historic reasons to do it but, wouldn't it be better to get CKAN as close to Flask as possible? While reviewing and cleaning code I notice that now that we are only supporting Flask we would win a lot of simplicity by making our app "as Flask as possible". 

Sometimes I don't see a noticeable feature that requires such layers of complexity on top of what Flask provides out of the box. There are complex classes like CKANTestApp with vague docstrings that don't provide any context for why is it necessary an extra abstraction (and complexity) layer on top of Flask's builtins.

I would suggest that getting CKAN closer to Flask would provide several benefits:
  - Easier for new developers to understand what's going on
  - Easier to extend since since our syntax and architecture will be as close as other Flasks applications as possible
  - Less code to maintain
  - We would leverage better all of Flask's built in features.
  - Easier to leverage Flask's bests practices

I understand this requires major effort but it would be nice to get and idea if that's possible. It will help a lot when taking technical decisions and technical roadmap.

Examples of what I have in mind with "getting closer to Flask":
 - Use Flask's command line tools instead of our custom cli module.
 - Deprecate the use of context in favor of Flask's g object.
 - Simplify our application architecture to better leverage the Application Factory.
 - Simplify our test suit by using native Flask's functions.

**Comment by tino097** [comment]
I agree about making CKAN more "flasky". One PR that is also to be taken in concideration is  #6560 .
